app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt

```diff
@@ -48,29 +43,14 @@ open class TabCounter @JvmOverloads constructor(
     }
 
     fun setCountWithAnimation(count: Int) {
-        updateContentDescription(count)
-
-        // Don't animate from initial state.
-        if (this.count == 0) {
-            setCount(count)
-            return
-        }
+        setCount(count)
```

Comment:
We can't fully remove the INTERNAL_COUNT == count case because now it's the opposite scenario, we animate almost always (including when we just press '+' from tab tray and go to home fragment, even though the number stays the same). 
I think we need something along the lines of: 
`when {`
`count == INTERNAL_COUNT -> return`
            `count == 0 -> {`
                `setCount(count)`
                `return`
            `} // Initial state.`
            `count > MAX_VISIBLE_TABS -> {`
                `setCount(count)`
                `return`
            `} // There are still over MAX_VISIBLE_TABS tabs open.`
        `}`

In other words, basically we need to do nothing in case it's equal, and to do also `setCount` in the other 2 cases. @hakkikaancaliskan What do you think?

Reply:
You know the best, I'm amateur. Is there something else with pr?

Reply:
When I look to the [current master](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/components/toolbar/TabCounter.kt) maybe we can take setCount to bottom of when.
```
when {
            count == INTERNAL_COUNT -> return // There isn't any tab added or removed.
            count == 0 -> return // Initial state.
            count > MAX_VISIBLE_TABS -> return // There are still over MAX_VISIBLE_TABS tabs open.
        }

        setCount(count)
```

What you think @BranescuMihai?

Reply:
Sorry, my bad.

Reply:
You know the best, I'm amateur. Fixing now as you said.

Reply:
Also I'm taking `setCount` to the bottom of `when`.
What do you think?

Reply:
![resim](https://user-images.githubusercontent.com/17825767/85319063-b8cb9f80-b4c9-11ea-8718-99471ef43c4a.png)
How is this? I've tested and working great.

```
// No need to animate on these cases.
        when {
            count == INTERNAL_COUNT -> {
                Log.d("kaan", "count == INTERNAL_COUNT")
                return
            } // There isn't any tab added or removed.
            INTERNAL_COUNT == 0 -> {
                Log.d("kaan", "INTERNAL_COUNT == 0")
                setCount(count)
                return
            } // Initial state.
            count > MAX_VISIBLE_TABS && INTERNAL_COUNT > MAX_VISIBLE_TABS -> {
                Log.d("kaan", "count > MAX_VISIBLE_TABS && INTERNAL_COUNT > MAX_VISIBLE_TABS")
                INTERNAL_COUNT = count
                return
            } // There are still over MAX_VISIBLE_TABS tabs open.
        }

        Log.d("kaan", "animate")
        setCount(count)
```